subworkflows/local/yaml_input/main.nf

```diff
@@ -83,3 +84,61 @@ workflow YAML_INPUT {
 def readYAML(yamlfile) {
     return new org.yaml.snakeyaml.Yaml().load(new FileReader(yamlfile.toString()))
 }
+
+def GET_VALIDATED_CHANNEL (data_type, tolid_ver, read_type, defined_class, project_id, data_files) {
+    // Based on the the functions added in commit: 61f4ad9
+    // Edited to be a function working on the raw yaml data
+    // rather than channels as it was previously
+
+    // Initialise defaults
+    def fofn_files = []
+    def direct_files = []
+    def files_list = data_files instanceof List ? data_files : [data_files]
+
+    // Process each file - separate FOFN from direct files
+        files_list.each { file_path ->
+            if (file_path.toString().contains(".fofn")) {
+                def fofn_content = file(file_path).text.split('\n')
+                    .collect { it.trim() }
+                    .findAll { it } // Remove empty lines
+                fofn_files.addAll(fofn_content)
+            } else {
+                direct_files.add(file_path)
+            }
+        }
+
+        // Combine all files
+        def all_files = direct_files + fofn_files
+
+        // Validate files based on data type
+        if (data_type == "cram") {
+            def invalid_files = all_files.findAll {
+                !it.toString().contains(".cram")
+            }
+            if (invalid_files.size() > 0) {
+                error "One of the input hic files does not match cram format. Invalid files: ${invalid_files}"
+            }
+        } else if (data_type == "longread") {
```

Comment:
I wonder if we be more specific about the read_type, ie. hifi, ont,clr and ccs?

Reply:
I don't think so in this case, at this point we are only checking for valid prefix to generate the channels.

Is there any alternate logic needed for longread subtypes?